x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlAuthenticator.java

```diff
@@ -236,6 +238,34 @@ private Assertion decrypt(EncryptedAssertion encrypted) {
         return attributes;
     }
 
+    private void checkAuthnStatement(List<AuthnStatement> authnStatements) {
+        if (authnStatements.size() != 1) {
+            throw samlException("SAML Assertion subject contains {} Authn Statements while exactly one was expected.", authnStatements
+                .size());
+        }
+        for (AuthnStatement authnStatement : authnStatements) {
+            // "past now" that is now - the maximum skew we will tolerate. Essentially "if our clock is 2min fast, what time is it now?"
+            final Instant now = now();
+            final Instant pastNow = now.minusMillis(maxSkewInMillis());
+            if (authnStatement.getSessionNotOnOrAfter() != null &&
+                pastNow.isBefore(toInstant(authnStatement.getSessionNotOnOrAfter())) == false) {
+                throw samlException("Rejecting SAML assertion's Authentication Statement because [{}] is on/after [{}]", pastNow,
+                    authnStatement.getSessionNotOnOrAfter());
+            }
+            List<String> reqAuthnCtxClassRef = this.getSpConfiguration().getReqAuthnCtxClassRef();
+            if (reqAuthnCtxClassRef.isEmpty() == false) {
+                String authnCtxClassRefValue = null;
+                if (authnStatement.getAuthnContext() != null && authnStatement.getAuthnContext().getAuthnContextClassRef() != null) {
+                    authnCtxClassRefValue = authnStatement.getAuthnContext().getAuthnContextClassRef().getAuthnContextClassRef();
+                }
+                if (Strings.isNullOrEmpty(authnCtxClassRefValue) || reqAuthnCtxClassRef.contains(authnCtxClassRefValue) == false) {
+                    throw samlException("Rejecting SAML assertion as the AuthnContextClassRef {} is not one of the ({}) that were " +
```

Comment:
Wrap the first message parameter (`{}`) as `[{}]` so that any spaces (or the empty string) are clearly delimited.